### PR TITLE
PB-37029 Delete Attachment File

### PR DESCRIPTION
**Pull request by PiotrRudnik-one**

### CODE REVIEW

File: sdk/Silanis.ESL.SDK/src/AttachmentFile.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+namespace Silanis.ESL.SDK
+{
+    public class AttachmentFile
+    {
+        public Int32 Id {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/AttachmentFile.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/AttachmentFile.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            get; set;
+        }
+
+        public DateTime InsertDate {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/AttachmentFile.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/AttachmentFile.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            get; set;
+        }
+
+        public String Name {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/AttachmentFile.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/AttachmentFile.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            get; set;
+        }
+
+        public Boolean Preview {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/AttachmentFile.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentFileConverter.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        public AttachmentFileConverter (ESL.SDK.AttachmentFile sdkAttachmentFile)
+        {
+            this.sdkAttachmentFile = sdkAttachmentFile;
+        }
+
+        public Silanis.ESL.API.AttachmentFile ToApiAttachmentFile()
+        {
+            if (sdkAttachmentFile == null) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentFileConverter.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentFileConverter.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            result.SetInsertDate(sdkAttachmentFile.InsertDate.Ticks);
+            result.Name = sdkAttachmentFile.Name;
+            result.Preview = sdkAttachmentFile.Preview;
+
+            return result;
+        }
+
+        public ESL.SDK.AttachmentFile ToSDKAttachmentFile ()
+        {
+            if (apiAttachmentFile == null) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentFileConverter.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentFileConverter.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                return sdkAttachmentFile;
+            }
+
+            ESL.SDK.AttachmentFile result = new SDK.AttachmentFile {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentFileConverter.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentFile.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+namespace Silanis.ESL.API
+{
+    public class AttachmentFile
+    {
+
+       
+        private DateTime insertDate;
+
+        [JsonProperty ("id")]
+        public Int32 Id {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentFile.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentFile.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+            get; set;
+        }
+
+        [JsonProperty ("name")]
+        public String Name {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentFile.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentFile.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            get; set;
+        }
+
+        [JsonProperty ("preview")]
+        public Boolean Preview {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentFile.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 11]
```diff
@@ -4,64 +4,67 @@
...
+    internal class AttachmentRequirement
     {
-                get; set;
+
+        // Fields
+        private IList<AttachmentFile> files = new List<AttachmentFile> ();
+        // Accessors
+
+        [JsonProperty ("comment")]
+        public String Comment {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 12]
```diff
@@ -4,64 +4,67 @@
...
-    
-		    
-    [JsonProperty("data")]
-    public IDictionary<string, object> Data
-    {
-                get; set;
+
+
+        [JsonProperty ("data")]
+        public IDictionary<string, object> Data {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 13]
```diff
@@ -4,64 +4,67 @@
...
-    
-		    
-    [JsonProperty("description")]
-    public String Description
-    {
-                get; set;
+
+
+        [JsonProperty ("description")]
+        public String Description {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 14]
```diff
@@ -4,64 +4,67 @@
...
-    
-		    
-    [JsonProperty("id")]
-    public String Id
-    {
-                get; set;
+
+
+        [JsonProperty ("id")]
+        public String Id {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 15]
```diff
@@ -4,64 +4,67 @@
...
-    
-		    
-    [JsonProperty("name")]
-    public String Name
-    {
-                get; set;
+
+
+        [JsonProperty ("name")]
+        public String Name {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 16]
```diff
@@ -4,64 +4,67 @@
...
-    
-		    
-    [JsonProperty("required")]
-    public Nullable<Boolean> Required
-    {
-                get; set;
+
+
+        [JsonProperty ("required")]
+        public Nullable<Boolean> Required {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 17]
```diff
@@ -4,64 +4,67 @@
...
-    
-		    
-    [JsonProperty("status")]
-    public string Status
-    {
-                get; set;
+
+
+        [JsonProperty ("status")]
+        public string Status {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 18]
```diff
@@ -4,64 +4,67 @@
...
+            get; set;
+        }
+
+        [JsonProperty ("files")]
+        public IList<AttachmentFile> Files {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 19]
```diff
@@ -4,64 +4,67 @@
...
+            get {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs [hunk 20]
```diff
@@ -4,64 +4,67 @@
...
+                return files;
+            }
+            set {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Models/AttachmentRequirement.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/RestClient.cs [hunk 21]
```diff
@@ -205,6 +205,24 @@ public string Get(string path, string acceptType)
             return response;
         }
 
+        public void Delete(string path, string jsonPayload, string sessionId)
+        {
+            support.LogRequest ("DELETE", path);
+
+            byte [] payloadBytes = null;
+            if (jsonPayload != null) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/RestClient.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/RestClient.cs [hunk 22]
```diff
@@ -205,6 +205,24 @@ public string Get(string path, string acceptType)
...
+                payloadBytes = Converter.ToBytes (jsonPayload);
+            } else {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/RestClient.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs [hunk 23]
```diff
@@ -198,6 +198,23 @@ public void UploadAttachment(PackageId packageId, string attachmentId, IDictiona
...
 
+        public void DeletaAttachmentFile(PackageId packageId, string attachmentId, Int32 fileId, string signerSessionId)
+        {
+            string path = template.UrlFor (UrlTemplate.DELETE_ATTACHMENT_FILE_PATH)
+                 .Replace ("{packageId}", packageId.Id)
+                 .Replace ("{attachmentId}", attachmentId)
+                 .Replace("{fileId}", fileId.ToString())
+                 .Build ();
+
+            try {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs [hunk 24]
```diff
@@ -198,6 +198,23 @@ public void UploadAttachment(PackageId packageId, string attachmentId, IDictiona
...
+                restClient.Delete (path, null, signerSessionId);
+            } catch (EslServerException e) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs [hunk 25]
```diff
@@ -198,6 +198,23 @@ public void UploadAttachment(PackageId packageId, string attachmentId, IDictiona
...
+                throw new EslServerException ("Could not delete attachment file for signer. Exception: " + e.Message, e.ServerError, e);
+            } catch (Exception e) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Services/Internal/AttachmentRequirementApiClient.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentRequirementConverter.cs [hunk 26]
```diff
@@ -81,12 +82,33 @@ public Silanis.ESL.SDK.AttachmentRequirement ToSDKAttachmentRequirement()
...
+            foreach (var file in sdkAttachmentRequirement.Files) {
+                files.Add (new AttachmentFileConverter (file).ToApiAttachmentFile ());
+            }
+            return files;
+        }
+
+        public IList<Silanis.ESL.SDK.AttachmentFile> GetSDKAttachmentFiles()
+        {
+            IList<Silanis.ESL.SDK.AttachmentFile> files = new List<Silanis.ESL.SDK.AttachmentFile> ();
+            foreach (var file in apiAttachmentRequirement.Files) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentRequirementConverter.cs]: Please put "{" on the following line, as per the C # code style guide.

File: sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentRequirementConverter.cs [hunk 27]
```diff
@@ -81,12 +82,33 @@ public Silanis.ESL.SDK.AttachmentRequirement ToSDKAttachmentRequirement()
...
+                return result;
 			}
 
 			return sdkAttachmentRequirement;
 		}
+
+        public IList<Silanis.ESL.API.AttachmentFile> GetApiAttachmentFiles()
+        {
+            IList <Silanis.ESL.API.AttachmentFile> files = new List<Silanis.ESL.API.AttachmentFile> ();
+            foreach (var file in sdkAttachmentRequirement.Files) {
```
Comment (SeungChoi) [inline sdk/Silanis.ESL.SDK/src/Internal/Conversion/AttachmentRequirementConverter.cs]: Please put "{" on the following line, as per the C # code style guide.

**Comment by PiotrRudnik-one** [comment]
I think I have fixed all places

**Pull request merged**